ios/RiveReactNativeView.swift

```diff
@@ -394,6 +394,24 @@ class RiveReactNativeView: RCTView, RivePlayerDelegate, RiveStateMachineDelegate
         viewModel?.setInput(inputName, value: value)
     }
     
+    func getBooleanState(inputName: String) -> Bool? {
+        return viewModel?.boolInput(named: inputName)?.value();
+    }
+    
+    func getNumberState(inputName: String) -> Float? {
+        return viewModel?.numberInput(named: inputName)?.value();
+    }
+
+    func getBooleanStateAtPath(inputName: String, path: String) -> Bool? {
+        let input = viewModel?.riveModel?.artboard?.getBool(inputName, path: path);
+        return input?.value();
+    }
+    
+    func getNumberStateAtPath(inputName: String, path: String) -> Float? {
+        let input = viewModel?.riveModel?.artboard?.getNumber(inputName, path: path);
+        return input?.value();
+    }
```

Comment:
Interesting that we don't expose these in `RiveViewModel`…

Reply:
It would be good to expose. We never did for any of the runtimes, except Web and Unity I believe.

Android doesn't have easy getters for normal or nested inputs.